engine/extensions-pentaho-metadata/source/org/pentaho/reporting/engine/classic/extensions/datasources/pmd/SimplePmdDataFactory.java

```diff
@@ -651,8 +652,19 @@ protected String computedQuery( final String queryName, final DataRow parameters
     retval.add( translateQuery( queryName ) );
     retval.add( domainId );
     retval.add( xmiFile );
-    retval.add( getContextKey() );
+    retval.add( getContextKeyParentIdentifier() );
```

Comment:
@tmorgner by using the root key identifier, I can insure a repeatable query hash. What do you think about this?